tests/unit/completion/test_completer.py

```diff
@@ -0,0 +1,136 @@
+# vim: ft=python fileencoding=utf-8 sts=4 sw=4 et:
+
+# Copyright 2016 Ryan Roden-Corrent (rcorre) <ryan@rcorre.net>
+#
+# This file is part of qutebrowser.
+#
+# qutebrowser is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# qutebrowser is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with qutebrowser.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Tests for the Completer Object."""
+
+from unittest.mock import Mock
+
+import pytest
+from PyQt5.QtGui import QStandardItemModel
+
+from qutebrowser.completion.completer import Completer
+from qutebrowser.utils.usertypes import Completion
+
+
+class FakeCompletionModel(QStandardItemModel):
+
+    """Stub for a completion model."""
+
+    DUMB_SORT = Mock()
+
+    def __init__(self, kind, parent=None):
+        super().__init__(parent)
+        self.kind = kind
+
+
+@pytest.fixture
+def cmd(stubs):
+    """Create the statusbar command prompt the completer uses."""
+    return stubs.FakeStatusbarCommand()
+
+
+@pytest.fixture
+def completer(qtbot, cmd, config_stub):
+    """Create the completer used for testing."""
+    config_stub.data = {'completion': {'auto-open': False}}
+    return Completer(cmd, 0)
+
+
+@pytest.fixture
```

Comment:
Since you don't actually need a reference to `instances` in any test, you could do `@pytest.fixture(autouse=True)` here (and maybe remove the `return instances` so it's automatically used without having to add it to the signature.